### PR TITLE
Grid scrolling bugs

### LINKED ISSUE #4391: Horizontal scrollbar is in the middle of the results pane when there are not many result rows
Issue Type: <b>Bug</b>

1. Run a query that returns enough content to scroll horizontally but not enough rows to scroll vertically, such as `select '1', 2, 'this is really long text this is really long text this is really long text this is really long text this is really long text this is really long text this is really long text this is really long text ', 'this is normal text'`

Actual behavior: The result set's horizontal scrollbar shows up in the middle of the results pane
Expected behavior: The scrollbar is in a more natural location, like SSMS where it shows up at the bottom of the results pane

Azure Data Studio version: azuredatastudio 1.5.1 (9ca6200018fc206d67a47229f991901a8a453781, 2017-12-15T12:00:00.000Z)
OS version: Darwin x64 18.2.0

Query with 1 long row showing the horizontal scrollbar in the middle of the results pane

### PR DESCRIPTION
**Pull request by anthonydresser**
I found theres some grid scrolling issues, where grids would disappear during scrolling. I found this is because the class scrollableSplitview wasn't maintaining an updated size during resizing.

fixes #4391

Also fixes the scrollbar bug.
image
image

### CODE REVIEW

File: src/sql/parts/query/editor/gridPanel.ts [hunk 1]
```diff
@@ -708,7 +708,7 @@ class GridTable<T> extends Disposable implements IView {
 	}
 
 	public get maximumSize(): number {
-		return Math.max(this.maxSize, ACTIONBAR_HEIGHT + BOTTOM_PADDING);
+		return Number.POSITIVE_INFINITY; // Math.max(this.maxSize, ACTIONBAR_HEIGHT + BOTTOM_PADDING);
```
Comment (MattIrv) [inline src/sql/parts/query/editor/gridPanel.ts]: commented code

File: src/sql/base/browser/ui/scrollableSplitview/scrollableSplitview.ts [hunk 2]
```diff
@@ -316,14 +315,6 @@ export class ScrollableSplitView extends HeightMap implements IDisposable {
...
 			this.distributeViewSizes();
 		}
-
-		// Re-render the views. Set lastRenderTop and lastRenderHeight to undefined since
-		// this isn't actually scrolling up or down
-		let scrollTop = this.lastRenderTop;
-		let viewHeight = this.lastRenderHeight;
-		this.lastRenderTop = undefined;
-		this.lastRenderHeight = undefined;
-		this.render(scrollTop, viewHeight);
```
Comment (MattIrv) [inline src/sql/base/browser/ui/scrollableSplitview/scrollableSplitview.ts]: What bug does this fix? Removing this block of code reintroduces part of the bug I fixed in #4007 where the split view wasn't getting rendered correctly when un-maximizing a grid that was previously maximized.

To reproduce, run a query with multiple result sets and maximize and then un-maximize one of them. Until you scroll enough only one of the result sets will show up, and not in the right location

Comment (anthonydresser) [reply src/sql/base/browser/ui/scrollableSplitview/scrollableSplitview.ts]: I've fixed the issue by applying a more direct fix. https://github.com/Microsoft/azuredatastudio/pull/4396/commits/b6df533c9b713728a7cedde9514d1f28fbbfdd61

Comment (MattIrv) [reply src/sql/base/browser/ui/scrollableSplitview/scrollableSplitview.ts]: That's a much better fix than mine, thanks

**Comment by anthonydresser** [comment]
I'll merge for testing.

**Pull request merged**